In [1]:
%reload_ext nb_black

<IPython.core.display.Javascript object>

In [2]:
import dask.array as da
import numpy as np

<IPython.core.display.Javascript object>

In [3]:
%%timeit
x = da.random.random((10000, 10000), chunks=(1000, 1000))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

561 ms ± 157 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


<IPython.core.display.Javascript object>

In [4]:
%%timeit
x = da.random.random((10000, 10000), chunks=(250, 250))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

2.43 s ± 308 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


<IPython.core.display.Javascript object>

Running the code with chunks = (250,250) takes around 2.43 seconds.

In [5]:
%%timeit
x = da.random.random((10000, 10000), chunks=(500, 500))
y = x + x.T
z = y[::2, 5000:].mean(axis=1)
z.compute()

930 ms ± 235 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


<IPython.core.display.Javascript object>

Running the code with chunks = (500,500) only took around 930 milliseconds to run. This executed faster because the chunks are larger therefore there are less chunks. Less chunks lead to less computing.